### PR TITLE
prometheus: Export queue wait time metric

### PR DESCRIPTION
**Pull request by marusak**
Generates output as:
```
# HELP queue_time Average queue waiting time
queue_time 234
```

### CODE REVIEW

File: generate-stats [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
```
Comment (martinpitt) [inline generate-stats:1]: How about calling that "prometheus-stats"? That's the specific [prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/), so it wouldn't hurt to name the file as such and also add this link as a comment?

File: generate-stats [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import time
+import sqlite3
+import argparse
+import logging
+
+# Seconds in an hour
+HOUR = 3400
```
Comment (martinpitt) [inline generate-stats:27]: Not quite :)

File: generate-stats [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    # Get average wait seconds in last N hours
+    rows = cursor.execute("""\
+            SELECT SUM(wait_seconds) / COUNT(*)
+            FROM TestRuns
+            WHERE time > ? ;""", (since, )).fetchall()
+
+    wait = rows[0][0] or 0
+    output += "# Average waiting time for all projects\n"
+    output += "cockpit_wait_time {0}".format(wait)
```
Comment (martinpitt) [inline generate-stats:55]: Nitpick -- could we call this `queue_time` for clarity? (Not waiting for the test to finish, or something else)

File: run-queue [hunk 4]
```diff
@@ -65,7 +65,8 @@ def consume_webhook_queue(channel, amqp):
         elif action == 'closed' and pull_request.get('merged', False):
             sha = pull_request['head']['sha']
             db = os.path.join(get_images_data_dir(), "test-results.db")
-            cmd = "./store-tests --db {0} --repo {1} {2}".format(db, repo, sha)
+            data = os.path.join(get_images_data_dir(), "test-stats")
```
Comment (martinpitt) [inline run-queue:68]: Let's call it /prometheus, that seems to be the convention.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Let's also call the commit like "Export queue wait time prometheus metric"? When I saw the commit message, I first thought you wrote a script to genenerate a mock sqlite database :grinning:

**Comment by marusak** [comment]
Thank you for the review. Updated.
I also updated first comment to show how it looks now.

**Review by martinpitt — APPROVED** [review]
Dakujem!

**Pull request merged**